### PR TITLE
Tab context menu actions

### PR DESCRIPTION
**Pull request by sarinali**
# Description

Finishes the context menu actions for the tabs 

## Bugzilla bug ID

**ID:** 1903800
**Link:** https://bugzilla.mozilla.org/show_bug.cgi?id=1903800

## Type of change

- [x] New Test

# How does this resolve / make progress on that bug?

finishes.

# Screenshots / Explanations

N/A

# Comments / Concerns

N/A

### CODE REVIEW

File: modules/browser_object_tabbar.py [hunk 1]
```diff
@@ -222,3 +224,14 @@ def scroll_on_all_tabs_menu(self, down=True, pixels=200) -> BasePage:
...
+    def wait_for_num_tabs(self, num_tabs: int) -> BasePage:
+        """
+        Waits for the driver.window_handles to be updated accordingly with the number of tabs requested
+        """
+        try:
+            WebDriverWait(self.driver, 10).until(
+                lambda _: len(self.driver.window_handles) == num_tabs
+            )
+        except TimeoutException:
+            logging.warn("Timeout waiting for the number of windows to be:", 1)
```
Comment (ben-c-at-moz) [inline modules/browser_object_tabbar.py:237]: Shouldn't that `1` be `num_tabs`?

File: modules/browser_object_tabbar.py [hunk 2]
```diff
@@ -222,3 +224,14 @@ def scroll_on_all_tabs_menu(self, down=True, pixels=200) -> BasePage:
             self.actions.move_by_offset(0, (sign * pixels))
             self.actions.release()
             self.actions.perform()
+
+    def wait_for_num_tabs(self, num_tabs: int) -> BasePage:
+        """
+        Waits for the driver.window_handles to be updated accordingly with the number of tabs requested
+        """
+        try:
+            WebDriverWait(self.driver, 10).until(
```
Comment (ben-c-at-moz) [inline modules/browser_object_tabbar.py:233]: you don't need a WDW object here, that already exists on the BasePage as `wait`. So, `self.wait.until(...)`

File: tests/menus/test_tab_context_menu_actions.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        close_other_tabs.click()
+
+        tabs.hide_popup("tabContextMenu")
+
+    # verify the number of tabs
+    logging.info("Trying to verify only 1 tab remains.")
+    tabs.wait_for_num_tabs(1)
+
+
+def test_reopen_closed_tab(driver: Firefox):
```
Comment (ben-c-at-moz) [inline tests/menus/test_tab_context_menu_actions.py:150]: Reopening a tab through context menus already exists as a test! (Sorry. (But also you did the PR-review for it.) 😄 )

**Review by ben-c-at-moz — CHANGES_REQUESTED** [review]
A couple things here, and one test reduplication 😬 sorry

File: modules/browser_object_tabbar.py (hunk 2 above)
Comment (sarinali) [reply modules/browser_object_tabbar.py:233]: fixed!

File: modules/browser_object_tabbar.py (hunk 1 above)
Comment (sarinali) [reply modules/browser_object_tabbar.py:237]: yep my bad

**Pull request merged**